src/events/Windows.cpp

```diff
@@ -687,10 +688,16 @@ void Events::listener_unmapWindow(void* owner, void* data) {
     g_pHyprOpenGL->makeWindowSnapshot(PWINDOW);
 
     // swallowing
-    if (PWINDOW->m_pSwallowed && g_pCompositor->windowExists(PWINDOW->m_pSwallowed)) {
+    if (PWINDOW->m_pSwallowed) {
```

Comment:
what about previouslySwallowed? Also reset this :P

Reply:
Replaced with a `bool`. Is there a better method to handle the hide and unhide in C++ ?

Reply:
why not check `isHidden()`? If swallowed != null and swallowed->ishidden(), it's swallowed, if it's not hidden, it's not

Reply:
I thought about this, but it could be hidden for other reasons, like if another window is fullscreen, no ?

Reply:
grouped windows are hidden